source/Tests/Testing/Unit/UnitTestClassAnalyzerFixture.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using NUnit.Framework;
+using Octopus.RoslynAnalyzers;
+using Verify = Tests.CSharpVerifier<Octopus.RoslynAnalyzers.Testing.Unit.UnitTestClassAnalyzer>;
+
+namespace Tests.Testing.Unit
+{
+    public class UnitTestClassAnalyzerFixture
```

Comment:
Just to clarify (related to above comment on namespace extension) - whats the distinction between unit test and integration test in this context? I'm aware the meta around testing (integration and unit in particular) is currently in flux over in Octopus Server. Should these be calling `container.InIntegrationTestNamespace()` or `container.InUnitTestNamespace()` (from a separate extension method)?

Its a subtle difference I admit. But if they are separate contexts, then using `inIntegrationTestNamespace` to access the UnitTest base class for the test could be misleading.

Reply:
Yeah it's a bit messy at the moment. Fortunately not in that much flux _I think_. Does need a bit of clarification (on my backlog).

My current understanding is that there is plan to merge the integration test and unit test projects into a single project, and just differentiate between the types using the base classes (as those base classes have all the traits that we need). Turns out we're double coping a lot of references assemblies at the moment having two test projects, so things would be a lot faster if we just had it in the one and filtered by traits.

So we currently have a `UnitTest` base type in `Octopus.IntegrationTests` as a bit if an intermediate until we can spend some time reorganising it all. As I said above, I think the extension just needs a better name that will make sense for now.

Once we can reference the real `IntegrationTest` and `UnitTest` types we'll be able to ditch this extension completely I think.

Reply:
👍 Yeah that actually make a lot more sense to me (since we are merging the integration and unit test projects). I like your idea - keep things structured the way they are, and then give the extension method the better name.